### ISSUE TITLE
Errors when trying to update from 2.0.4

### ISSUE THREAD

**Issue by DukeLupus** [DukeLupus/dlFilter#73]
```
* /rename: unable to rename 'C:\Program Files (x86)\mIRC\dlFilter.mrc' to 'C:\Program Files (x86)\mIRC\dlFilter.mrc.v204' (line 4157, dlFilter.mrc)
-
[07:09] [dlFilter] Checking for dlFilter updates...
[07:09] [dlFilter] You can update to production version 2.07 - Significant fixes
[07:09] [dlFilter] Downloading new version of dlFilter...
-
* /bwrite: unable to open file 'C:\Program Files (x86)\mIRC\dlFilter.mrc.new' (line 4143, dlFilter.mrc)
```

**Comment by Sophist-UK** [comment]
I will address this issue in the next version. In the mean time please delete the following two files and try to update again:
```
C:\Program Files (x86)\mIRC\dlFilter.mrc.new
C:\Program Files (x86)\mIRC\dlFilter.mrc.v204
```

**Comment by Sophist-UK** [comment]
Thank you for reporting this. There are two possible causes:

a. That the `C:\Program Files (x86)\mIRC` directory is not writeable - please try to run mIRC temporarily with Administrative Privileges and see if it solves it. (The long term solution is to unload dlFilter, move it to a directory which is writeable without administrative privileges and reload it from there.) IMO this is the most likely cause. I will see if I can code for this.

b. That the script doesn't check for the existence of files before renaming / writing files - I haven't had the chance to check the code, but if I haven't already coded for this the fix is reasonably easy and I will address this issue in the next version. In the mean time please check whether the following files exist and if so delete them and try to update again:
```
C:\Program Files (x86)\mIRC\dlFilter.mrc.new
C:\Program Files (x86)\mIRC\dlFilter.mrc.v204
```

**Comment by Sophist-UK** [comment]
I have checked the code. I have already coded for existence of a saved file for the /rename, so it must be issue a. you are experiencing. However I do need to code for non-writable directory and for file being non-writable when directory is and give warning messages if we cannot do rename / write.

**Comment by Sophist-UK** [comment]
Hmm - user used GitReports to log the issue and didn't give his nick or email address - so no way to contact him to let him know how to resolve this.

**Comment by Sophist-UK** [comment]
Fixed in 7c6c350119c3ebfe6b10296ed11b3754989bf322
